packages/nuqs/src/parsers.ts

```
@@ -211,59 +195,46 @@ export function createMultiParser<T>(
-        ...this,
-        ...options
-      }
+      return { ...this, ...options }
     }
   }
 }
 
 // Parsers implementations -----------------------------------------------------
 
+// NaN-check helper: works for numbers and Dates (via unary + coercion).
+const ok = <T extends number | Date>(v: T): T | null => (+v == +v ? v : null)
```

Biome (2.4.15)

[error] This comparison uses the same expression on both sides.

(lint/suspicious/noSelfCompare)